src/priority-queue.ts

```diff
@@ -0,0 +1,257 @@
+import {ArmorCollection} from './collection';
+import {ArmorCollectionSelector} from './selector';
+import ArmorPriorityQueueComparator from './priority-queue-comparator';
+import ArmorPriorityQueueNodeChildren from './priority-queue-children';
+import ArmorPriorityQueueOptions from './priority-queue-options';
+
+export default class ArmorPriorityQueue<T> implements ArmorCollection<T> {
+	public elements: T[];
```

Comment:
- [ ] Create additional ArmorPriorityQueueState<T> interface here as property with name `state`. 

- [ ] Move elements property from pq to pq state class.

- [ ] Add a stringify method to pq class. We'll use it to serialize queue data for storage.

- [ ] Rename usages of `this.elements` to `this.state.elements`

```typescript
// Simple serialize example
public serialize(o: any): string | null {
   let result = null;
   try {
      result = JSON.stringify(this.state);
   } catch (e) {
      result = null;
   }

   return result;
}
```

- [ ] Add a parse method to pq class. 
```typescript
// Simple parse example
public parse(o: string): ArmorPriorityQueueState<T> | null {
   if (typeof o !== 'string' || o === '') {
      return null;
   }
   let result: ArmorPriorityQueue<T> | null = null;
   try {
      result = JSON.parse(o);
  } catch (e) {
      result = null;
  } 

  return result;
}

Create a new interface called ArmorPriorityQueueState<T> to hold instance related data like elements. It will be used for serialization later. That way we can serialize only the queue data and not the code or other functions which won't change.
The state instance hold variables related to the queue's _current state_. State needs one variable currently: `public elements: T[];`